### PR TITLE
fix: Fixed compilation after API change of TimerBase::execute

### PR DESCRIPTION
**Pull request by jmachowinski**
This is a followup of https://github.com/ros2/rclcpp/pull/2343

### CODE REVIEW

**Review by fujitatomoya — APPROVED** [review]
lgtm

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20659/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14990/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21362/)

File: rclcpp/topics/minimal_subscriber/time_triggered_wait_set_subscriber.cpp [hunk 1]
```diff
@@ -71,7 +71,9 @@ class TimeTriggeredWaitSetSubscriber : public rclcpp::Node
         case rclcpp::WaitResultKind::Ready:
           {
             if (wait_result.get_wait_set().get_rcl_wait_set().timers[0U]) {
-              timer_->execute_callback();
+              if (auto data = timer_->call()) {
+                timer_->execute_callback(*data);
```
Comment (fujitatomoya) [inline rclcpp/topics/minimal_subscriber/time_triggered_wait_set_subscriber.cpp:75]: Compile error.

```suggestion
                timer_->execute_callback(data);
```

File: rclcpp/wait_set/src/wait_set_topics_and_timer.cpp [hunk 2]
```diff
@@ -71,8 +71,10 @@ int32_t main(const int32_t argc, char ** const argv)
     const auto wait_result = wait_set.wait(2s);
     if (wait_result.kind() == rclcpp::WaitResultKind::Ready) {
       if (wait_result.get_wait_set().get_rcl_wait_set().timers[0U]) {
-        // The timer callback is executed manually here
-        one_off_timer->execute_callback();
+        if (auto data = one_off_timer->call()) {
+          // The timer callback is executed manually here
+          one_off_timer->execute_callback(*data);
```
Comment (fujitatomoya) [inline rclcpp/wait_set/src/wait_set_topics_and_timer.cpp:76]: Compile error.

```suggestion
          one_off_timer->execute_callback(data);
```

**Review by fujitatomoya — CHANGES_REQUESTED** [review]

**Review by fujitatomoya — APPROVED** [review]

**Comment by fujitatomoya** [comment]
@jmachowinski i will wait for https://github.com/ros2/rcl/pull/1142.

and then, i will start the CI with https://github.com/ros2/rclcpp/pull/2343(https://gist.githubusercontent.com/fujitatomoya/abec3a5908d5af11b95982106feace2c/raw/1b865510acff5b3d4fa57e81b45523b2d2c59b9d/ros2.repos)

CC: @clalancette @mjcarroll @wjwwood @alsora

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20761/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15060/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21440/)

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**